src/librbd/cache/pwl/rwl/WriteLog.cc

```diff
@@ -439,6 +439,19 @@ void WriteLog<I>::load_existing_entries(DeferredContexts &later) {
   this->update_sync_points(missing_sync_points, sync_point_entries, later, MIN_WRITE_ALLOC_SIZE);
 }
 
+template <typename I>
+void WriteLog<I>::calc_allocated_and_cached_data(std::shared_ptr<pwl::GenericLogEntry> log_entry, uint32_t alloc_size) {
```

Comment:
Does `alloc_size` need to be a parameter?  It is `MIN_WRITE_ALLOC_SIZE` for rwl mode and `MIN_WRITE_ALLOC_SSD_SIZE` for ssd mode and you are passing it through a virtual into a mode-specific override.  I would drop the parameter and just use the respective constant.